genie/st_transformer.py

```diff
@@ -28,28 +23,16 @@ def forward(self, x: Tensor) -> Tensor:
 
 class STBlock(nn.Module):
     # See Figure 4 of https://arxiv.org/pdf/2402.15391.pdf
-    def __init__(
-        self,
-        dim: int,
-        num_heads: int,
-        mlp_ratio: float = 4.0,
-        qkv_bias: bool = False,
-        proj_bias: bool = True,
-        ffn_bias: bool = True,
-        drop: float = 0.0,
-        attn_drop: float = 0.0,
-        qkv_norm: bool = True,
-    ) -> None:
+    def __init__(self, config) -> None:
         super().__init__()
-        self.norm1 = nn.Identity() if qkv_norm else nn.LayerNorm(dim, eps=1e-05)
-        # sequence dim is over each frame's 32x32 patch tokens
-        self.spatial_attn = SelfAttention(dim, num_heads=num_heads, qkv_bias=qkv_bias, proj_bias=proj_bias, attn_drop=attn_drop, qkv_norm=qkv_norm)
+        self.norm1 = nn.Identity() if config.qk_norm else nn.LayerNorm(config.d_model, eps=1e-05)
+        # sequence dim is over each frame's 20x20 patch tokens
+        self.spatial_attn = SelfAttention(config)
         # sequence dim is over time sequence (16)
-        self.temporal_attn = SelfAttention(dim, num_heads=num_heads, qkv_bias=qkv_bias, proj_bias=proj_bias, attn_drop=attn_drop, qkv_norm=qkv_norm)
+        self.temporal_attn = SelfAttention(config)
```

Comment:
let's have a default constructor like we had before and if you want to use config, add classmethod `from_config`